### ISSUE TITLE
Example hello_world.py keeps pending.

### ISSUE THREAD

**Issue by sakaia** [kubeflow/pipelines#3570]
### What steps did you take:
create microk8s1.15 + kubeflow1.0.2(istio)

```bash
$ git clone https://github.com/kubeflow/pipelines
$ cd pipelines//samples/core/helloworld
$ kubectl create serviceaccount pipeline-runner
$ python3 hello_world.py
$ kubectl create -f hello_world.py.yaml
$ kubectl describe workflow.argoproj.io/my-first-pipeline-v5v4j
```

### What happened:
The deployment keeps pending. Is there any way to solve this problem?

```bash
$ kubectl describe workflow.argoproj.io/my-first-pipeline-v5v4j
Name:         my-first-pipeline-v5v4j
Namespace:    default
Labels:       workflows.argoproj.io/phase=Running
Annotations:  pipelines.kubeflow.org/pipeline_spec: {"description": "A hello world pipeline.", "name": "My first pipeline"}
API Version:  argoproj.io/v1alpha1
Kind:         Workflow
Metadata:
  Creation Timestamp:  2020-04-21T02:41:32Z
  Generate Name:       my-first-pipeline-
  Generation:          3
  Resource Version:    74049
  Self Link:           /apis/argoproj.io/v1alpha1/namespaces/default/workflows/my-first-pipeline-v5v4j
  UID:                 dff5a35f-f455-47fd-8f31-b5f3ad6aacaf
Spec:
  Arguments:
  Entrypoint:            my-first-pipeline
  Service Account Name:  pipeline-runner
  Templates:
    Container:
      Args:
        echo "hello world"
      Command:
        sh
        -c
      Image:  library/bash:4.4.23
      Name:
      Resources:
    Inputs:
    Metadata:
    Name:  echo
    Outputs:
    Dag:
      Tasks:
        Arguments:
        Name:      echo
        Template:  echo
    Inputs:
    Metadata:
    Name:  my-first-pipeline
    Outputs:
Status:
  Finished At:  <nil>
  Nodes:
    my-first-pipeline-v5v4j:
      Children:
        my-first-pipeline-v5v4j-2832218302
      Display Name:   my-first-pipeline-v5v4j
      Finished At:    <nil>
      Id:             my-first-pipeline-v5v4j
      Name:           my-first-pipeline-v5v4j
      Phase:          Running
      Started At:     2020-04-21T02:41:32Z
      Template Name:  my-first-pipeline
      Type:           DAG
    my-first-pipeline-v5v4j-2832218302:
      Boundary ID:    my-first-pipeline-v5v4j
      Display Name:   echo
      Finished At:    <nil>
      Id:             my-first-pipeline-v5v4j-2832218302
      Message:        ContainerCreating
      Name:           my-first-pipeline-v5v4j.echo
      Phase:          Pending
      Started At:     2020-04-21T02:41:32Z
      Template Name:  echo
      Type:           Pod
  Phase:              Running
  Started At:         2020-04-21T02:41:32Z
Events:               <none>
```
Container logs are follows

```bash
$ kubectl logs my-first-pipeline-v5v4j-2832218302 -c wait
Error from server (BadRequest): container "wait" in pod "my-first-pipeline-v5v4j-2832218302" is waiting to start: ContainerCreating
$ kubectl logs my-first-pipeline-v5v4j-2832218302 -c main
Error from server (BadRequest): container "main" in pod "my-first-pipeline-v5v4j-2832218302" is waiting to start: ContainerCreating
```

### What did you expect to happen:
The deployment works successfully.

### Environment:

How did you deploy Kubeflow Pipelines (KFP)?

I install microk8s1.15 + kubeflow 1.0.2 (istio).
In my environment, tfjob and katib works fine. 

KFP version: 

KFP SDK version: 

kfp                                   0.4.0
kfp-server-api                        0.3.0

### Anything else you would like to add:

/kind bug
/area samples

**Comment by Ark-kun** [comment]
This is probably an issue with an underlying system (Argo or even Kubernetes)

Can you please try
```
kubectl describe pods my-first-pipeline-v5v4j-2832218302
kubectl get pods my-first-pipeline-v5v4j-2832218302 --output=yaml
```

**Comment by sakaia** [comment]
Thank you for your help. The outputs are already written above. I also find the warning on events.
But above output `Events:` says `<none>`

```bash
$ kubectl get events
LAST SEEN   TYPE      REASON        OBJECT                                       MESSAGE
50s         Warning   FailedMount   pod/my-first-pipeline-wgkg2-3423630397       MountVolume.SetUp failed for volume "docker-sock" : hostPath type check failed: /var/run/docker.sock is not a socket file
11m         Warning   FailedMount   pod/my-first-pipeline-wgkg2-3423630397       Unable to mount volumes for pod "my-first-pipeline-wgkg2-3423630397_default(b644c24f-2181-45a3-8b45-6de3e6778781)": timeout expired waiting for volumes to attach or mount for pod "default"/"my-first-pipeline-wgkg2-3423630397". list of unmounted volumes=[docker-sock]. list of unattached volumes=[podmetadata docker-sock pipeline-runner-token-4rghz]
```

**Comment by sakaia** [comment]
From seeing the error,  It seems following issue. https://github.com/kubeflow/pipelines/issues/561

Continue digging.

**Comment by sakaia** [comment]
This problem was solved. My environment is microk8s 1.15 + kubeflow 1.0.2

The problem was already documented on  https://github.com/argoproj/argo/issues/2557#issuecomment-607239438

I changed from `docker` to `k8sapi` for attribute `containerRuntimeExecutor`

```yaml
containerRuntimeExecutor: k8sapi
```
More specific, I edit `params.env` which is created  by `kfctl build` command
then execute "kfctl apply`

Of course upto microk8s 1.13, it uses docker but it changed to containerd from microk8s 1.14.
So symbolic link avoidance for docker is not effective.
